src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,167 @@
+
 package core.basesyntax;
 
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_FOR_EXCEPTION = "Index: ";
+    private static final String SIZE_FOR_EXCEPTION = ", Size: ";
+    private static final int NUMBER_TO_SUBTRACT = 1;
+    private static final int ZERO = 0;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < ZERO || index > size) {
+            throw new IndexOutOfBoundsException(INDEX_FOR_EXCEPTION
+                    + index + SIZE_FOR_EXCEPTION + size);
+        }
+        Node<T> newNode = new Node<>(value);
+        addNode(newNode, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndexForGet(index);
+        Node<T> node = getNode(index);
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        validateIndexForGet(index);
+        Node<T> node = getNode(index);
```

Comment:
you already have this check in `getNode()`
```suggestion
        Node<T> node = getNode(index);
```